lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/GradebookIfc.java

```diff
@@ -114,7 +116,20 @@ public boolean updateExternalAssessmentScore(final String gradebookUid, final St
 
 	public boolean updateExternalAssessmentScores(final String gradebookUid, final String siteId, final String externalId, final Map studentUidsToScores) {
 		try {
-			gradingService.updateExternalAssessmentScoresString(gradebookUid, siteId, externalId, studentUidsToScores);
+			Map<String, String> normalizedScores = null;
+			if (studentUidsToScores != null) {
+				normalizedScores = new HashMap<>();
+				for (Object entryObject : studentUidsToScores.entrySet()) {
+					Map.Entry<?, ?> entry = (Map.Entry<?, ?>) entryObject;
+					Object key = entry.getKey();
+					if (!(key instanceof String)) {
+						continue;
+					}
+					Object value = entry.getValue();
```

Comment:
if the Map had types this could be simplified

Reply:
done